programmers/로또의-최고-순위와-최저-순위.js

```diff
@@ -1,8 +1,8 @@
 function solution(lottos, win_nums) {
-  var answer = lottos.reduce((acc, val) => {
+  let answer = lottos.reduce((acc, val) => {
     return acc + win_nums.includes(val);
   }, 0);
```

Comment:
와우 정말 너무 잘하세요 ㅋㅋㅋㅋ
다만 굳이 똑같은 배열을 두번 반복시켜 줄 필요는 없어요.
reduce로 반복할 때 ```if```를 사용하여 필터효과를 같이 가져갈 수 있습니다.

```suggestion
  let zeros = 0;
  let answer = lottos.reduce((acc, val) => {
    if (val === 0) zeros += 1;
    return acc + win_nums.includes(val);
  }, 0);
```

Reply:
그치만 사실 이것도 매번 zeros를 비교해주기 때문에 배열의 개수만큼 비교해야되는 단점이 있지요.
아래처럼 존재여부를 먼저 검사하고 존재하지 않을 때만 0인지를 비교해주면 훨씬 효율적일거예요.
다만, 해당 문제는 배열의 개수가 6개로 제한되어있으니 이런건 신경쓰지 않아도 되겠죠 ㅋㅋㅋㅋ 
```suggestion
  let zeros = 0;
  let answer = lottos.reduce((acc, val) => {
    if (win_nums.includes(val)) acc += 1;
    else if (val === 0) zeros += 1;
    return acc;
  }, 0);
```

Reply:
저 같은 경우는 이렇게 풀었어요.
Key와 Value의 타입은 시간복잡도가 O(1)의 효율을 같기 때문에 상당히 빠르답니다.
아래의 lankings 오브젝트로 정답이 맞춘개수가 6개면 1등을 반환하는 구조에요 :D
```suggestion
function solution(lottos, win_nums) {
    const lankings = {
        6: 1, 5: 2, 4: 3, 
        3: 4, 2: 5, 1: 6, 
        0: 6
    };
    
    let [sameCnt, zeroCnt] = [0, 0];
    for (let lotto of lottos) {
        if (win_nums.includes(lotto)) sameCnt += 1;
        else if (lotto === 0) zeroCnt += 1;
    }
    
    return [lankings[sameCnt + zeroCnt], lankings[sameCnt]];
}
```